File: paper/src/main/java/one/hyro/paper/commands/MinigamesCommand.java
package one.hyro.paper.commands;

import io.papermc.paper.command.brigadier.BasicCommand;
import io.papermc.paper.command.brigadier.CommandSourceStack;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import one.hyro.builders.GameMenu;
import one.hyro.builders.CustomItem;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class MinigamesCommand implements BasicCommand {
    @Override
    public void execute(@NotNull CommandSourceStack stack, @NotNull String[] args) {
        if (!(stack.getExecutor() instanceof Player player)) return;

        CustomItem survival = new CustomItem(Material.NOTE_BLOCK)
                .setCustomId("survival")
                .setDisplayName(Component.text("Survival", NamedTextColor.GREEN))
                .onClick(clicker -> {
                    clicker.closeInventory();
                    clicker.sendMessage(Component.text("Coming soon!"));
                })
                .build();

        GameMenu minigames = new GameMenu()
                .setCustomId("minigames")
                .setTitle(Component.text("Minigames"))
                .setSize(27)
                .setHolder(player)
                .setItem(13, survival)
                .build();

        player.openInventory(minigames.getInventory());
    }
}

File: core/src/main/kotlin/commands/MinigamesCommand.kt
package one.hyro.commands

import io.papermc.paper.command.brigadier.BasicCommand
import io.papermc.paper.command.brigadier.CommandSourceStack
import net.kyori.adventure.text.Component
import one.hyro.builder.inventory.CustomItem
import one.hyro.builder.inventory.Menu
import org.bukkit.Material
import org.bukkit.entity.Player

object MinigamesCommand: BasicCommand {
    override fun execute(stack: CommandSourceStack, args: Array<out String>) {
        if (stack.executor !is Player) return

        val survival = CustomItem(Material.GRASS_BLOCK)
            .displayName(Component.text("Survival"))
            .lore(Component.text("Survive in a world with limited resources"))
            .click { player: Player ->
                player.sendMessage(Component.text("You clicked on the survival minigame!"))
            }
            .build()

        val duels = CustomItem(Material.DIAMOND_SWORD)
            .displayName(Component.text("Duels"))
            .lore(Component.text("Fight against other players"))
            .click { player: Player ->
                player.sendMessage(Component.text("You clicked on the duels minigame!"))
            }
            .build()

        val minigames = Menu()
            .title(Component.text("Minigames"))
            .item(12, survival)
            .item(14, duels)
            .build()

        val player: Player = stack.executor as Player
        player.openInventory(minigames.inventory)
    }
}